ui-react/packages/atlasmap-ui/src/CanvasView/components/FieldsBoxHeader.tsx

```diff
@@ -46,89 +46,98 @@ export const FieldsBoxHeader: FunctionComponent<IFieldsBoxHeaderProps> = ({
   return (
     <Stack>
       <StackItem>
-        <Level>
-          <LevelItem>{title}</LevelItem>
-          <LevelItem>
-            <Dropdown
-              toggle={
-                <DropdownToggle
-                  splitButtonItems={[
-                    <DropdownToggleAction key="action" onClick={toggleSearch}>
-                      <SearchIcon />
-                    </DropdownToggleAction>,
-                  ]}
-                  splitButtonVariant={'action'}
-                  onToggle={toggleActions}
+        <Flex breakpointMods={[{modifier: FlexModifiers['space-items-none']}]}>
+          {showSearch ? (
+              <FlexItem>{title}</FlexItem>
+          ) : (
+            <FlexItem breakpointMods={[{modifier: FlexModifiers['flex-1']}]}>{title}</FlexItem>
+          )}
+          <FlexItem>
+            <Button
+              variant={ButtonVariant.control}
+              aria-label="Search button for search input"
+              onClick={toggleSearch}
+            >
+              <SearchIcon />
+            </Button>
+          </FlexItem>
+          {showSearch && (
+            <FlexItem breakpointMods={[{modifier: FlexModifiers['flex-1']}]}>
+              <InputGroup>
+                <TextInput
+                  name={'source-search'}
+                  id={'source-search'}
+                  type="search"
+                  aria-label="Search source fields"
+                  autoFocus={true}
+                  onChange={onSearch}
                 />
-              }
+              </InputGroup>
+            </FlexItem>
+          )}
+          <FlexItem breakpointMods={[{modifier: FlexModifiers['align-self-flex-end']}]}>
+            <Dropdown
+              toggle={<DropdownToggle onToggle={toggleActions} />}
               isOpen={showActions}
               position={'right'}
               dropdownItems={[
                 <DropdownItem variant={'icon'} key={'import'}>
-                  <DropdownItemIcon>
-                    <ImportIcon />
-                  </DropdownItemIcon>
-                  <Tooltip
-                    position={'auto'}
-                    enableFlip={true}
-                    content={<div>Import instance or schema file</div>}
-                  >
-                    <FilePicker
-                      extensions={['json', 'xml', 'xsd']}
-                      onChange={(selectedFile: File) => onImport(selectedFile)}
-                      onError={(errMsg: any) => console.error(errMsg)}
-                    >
-                      <div>Import</div>
-                    </FilePicker>
-                  </Tooltip>
+                    <SplitItem isFilled>
```

Comment:
Dropdown items with an icon and text is a layout that's supported by the DropdownItem component already, we should try to make the FilePicker behave with that instead of handling the layout bit ourself.

Did you try moving the FilePicker to be the first child of the DropdownItem? Something like:

```tsx
<DropdownItem variant={'icon'} key={'import'}>
  <FilePicker
    extensions={['json', 'xml', 'xsd']}
    onChange={(selectedFile: File) => onImport(selectedFile)}
    onError={(errMsg: any) => console.error(errMsg)}
  >
    <>
      <DropdownItemIcon>
        <ImportIcon />
      </DropdownItemIcon>
      <Tooltip
        position={'auto'}
        enableFlip={true}
        content={<div>Import instance or schema file</div>}
      >
          <div>Import</div>
      </Tooltip>
    </>
  </FilePicker>
</DropdownItem>
```

Reply:
Yes, this changes was fairly dependent on my previous one, which hadn't been approved yet, so I was hoping GH would be smart enough to auto-remove it once it saw the same commit merged in the codebase. Apparently not, so I'll re-push this after updating things locally now that the other one has been merged.